chromium_src/components/content_settings/core/common/cookie_settings_base.cc

```diff
@@ -292,13 +292,19 @@ CookieSettingsBase::GetCurrentCookieSettingWithBraveMetadata() {
 // were explicit. We use explicit setting to enable 1PES mode, but in this mode
 // we still want to block 3p frames as usual and not fallback to "allow
 // everything" path.
+#define BRAVE_COOKIE_SETTINGS_BASE_DECIDE_ACCESS                              \
```

Comment:
seems like it would be better to rename `CookieSettingsBase` -> `CookieSettingsBaseChromiumImpl` and then create `CookieSettingsBase` that subclasses `CookieSettingsBaseChromiumImpl`?

Reply:
I'm also a bit concerned about the complexity of this and not sure why it's so complex, but I guess that's a follow-up issue. Are there tests that cover all of this?

Reply:
I'll let @cdesouza-chromium respond regarding cleaning up the override, but yes we do have pretty comprehensive test coverage for this change with upstream tests (those are what led us to this solution).

Reply:
These overrides are changing significantly in `cr123`, and @goodov has in fact fixed some of of the fixes we did in `cr123`.

Reply:
@goodov that makes sense for the complexity of the check, but still seems like the override could eliminate the need for patching here?

Reply:
we can discuss if there should be a follow-up for this patch or not